Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,282 @@
+
+* `fixedField_fixingSubgroup` : the fixing field of the
+  fixing subgroup of `L` is equal to `L` itself.
+
+For any subgroup of `Gal(K/k)` `H` :
+
+* `restrict_fixedField` : For a Galois intermediate field `M`, the fixed field of the image of `H`
+  restricted to `M` is equal to the fixed field of `H` intersected with `M`.
+* `fixingSubgroup_fixedField` : If `H` is closed, the fixing subgroup of the fixed field of `H`
+  is equal to `H` itself.
+
+The fundamental theorem of infinite galois theory :
+
+* `IntermediateFieldEquivClosedSubgroup` : The order equivalence is given by mapping any
+  intermediate field `L` to the fixing subgroup of `L`, with its inverse mapping any
+  closed subgroup of `Gal(K/k)` `H` to the fixed field of `H`. The composition is equal to
+  the identity as described in the lemmas above, and compatibility with the order follows easily.
+
+Special cases :
+
+* `OpeniffFinite` : The fixing subgroup of an intermediate field `L` is open iff
+  `L` is finite-dimensional.
+
+* `NormaliffGalois` : The fixing subgroup of an intermediate field `L` is normal iff
+  `L` is Galois.
+
+-/
+
+variable {k K : Type*} [Field k] [Field K] [Algebra k K]
+
+namespace InfiniteGalois
+
+open Pointwise FiniteGaloisIntermediateField AlgEquiv
+--Note: The `adjoin`s below are `FiniteGaloisIntermediateField.adjoin`
+
+instance : TopologicalSpace (K ≃ₐ[k] K) := inferInstance
+
+lemma fixingSubgroup_isClosed (L : IntermediateField k K) [IsGalois k K] :
+    IsClosed (L.fixingSubgroup : Set (K ≃ₐ[k] K)) where
+    isOpen_compl := isOpen_iff_mem_nhds.mpr fun σ h => by
+      apply mem_nhds_iff.mpr
+      rcases Set.not_subset.mp ((mem_fixingSubgroup_iff (K ≃ₐ[k] K)).not.mp h) with ⟨y,yL,ne⟩
+      use σ • ((adjoin k {y}).1.fixingSubgroup : Set (K ≃ₐ[k] K))
+      constructor
+      · intro f hf
+        rcases (Set.mem_smul_set.mp hf) with ⟨g,hg,eq⟩
+        simp only [Set.mem_compl_iff, SetLike.mem_coe, ← eq]
+        apply (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).not.mpr
+        push_neg
+        use y
+        simp only [yL, smul_eq_mul, AlgEquiv.smul_def, AlgEquiv.mul_apply, ne_eq, true_and]
+        have : g y = y := (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp hg y <|
+          adjoin_simple_le_iff.mp fun ⦃x⦄ a ↦ a
+        simpa only [this, ne_eq, AlgEquiv.smul_def] using ne
+      · simp only [(IntermediateField.fixingSubgroup_isOpen (adjoin k {y}).1).smul σ, true_and]
+        use 1
+        simp only [SetLike.mem_coe, smul_eq_mul, mul_one, and_true, Subgroup.one_mem]
+
+lemma fixedField_fixingSubgroup (L : IntermediateField k K) [IsGalois k K] :
+    IntermediateField.fixedField L.fixingSubgroup = L := by
+  letI : IsGalois L K := inferInstance
+  apply le_antisymm
+  · intro x hx
+    rw [IntermediateField.mem_fixedField_iff] at hx
+    have mem : x ∈ (adjoin L {x}).1 := subset_adjoin _ _ rfl
+    have : IntermediateField.fixedField (⊤ : Subgroup ((adjoin L {x}) ≃ₐ[L] (adjoin L {x}))) = ⊥ :=
+      (IsGalois.tfae.out 0 1).mp (by infer_instance)
```

Comment:
`inferInstance` is the term, if you want to avoid going into tactic mode (probably saves a few milliseconds ;-) )

Reply:
I doesn't know clearly why `inferInstance` failed here, it doesn't seem to work.

Reply:
Yes that is surprising, `(inferInstance : IsGalois ↥L ↥(adjoin ↥L {x}).toIntermediateField)` works so it must be something to do with the order in which things are unified and metavariables are solved for.